controllers/process.go

```diff
@@ -212,16 +212,24 @@ func refreshEnvoyFilter(instance *microservicev1alpha2.SmartLimiter, r *SmartLim
 			return reconcile.Result{},nil
 		}
 
+		foundSpec,err := json.Marshal(found.Spec)
+		if err != nil {
+			log.Errorf("marshal found.spec err: %+v",err)
+		}
+		objSpec,err := json.Marshal(obj.Spec)
+		if err != nil {
+			log.Errorf("marshal obj.spec err: %+v",err)
+		}
 		// spec is not nil , update
 		if obj.Spec != nil {
-			if !reflect.DeepEqual(found.Spec, obj.Spec) {
-				log.Infof("Update a new EnvoyFilter,%v", loc)
+			if !reflect.DeepEqual(string(foundSpec), string(objSpec)) {
```

Comment:
We'd better make clear why deepequals not works if have time.

Reply:
compare byte slice is enough. no need to convert to string to have a dup mem cost.